### PR TITLE
Remove possible injection points and prevent future injection with CodeQL.

### PR DESCRIPTION
**Pull request by TreyE**
Fix several possible code injection locations and introduce tools to prevent the addition of new ones.

Also adds a framework for incremental correction and prevention of existing CodeQLfailures.

# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [X] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [X] For all UI changes, there is cucumber coverage
- [X] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [X] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [X] There are no inline styles added
- [X] There are no inline javascript added
- [X] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [X] Code does not use .html_safe
- [X] All images added/updated have alt text
- [X] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix 
- [X] Security
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [X] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187423479

# A brief description of the changes

Current behavior:  Multiple locations allow or risk code injection as discovered by CodeQL analysis.

New behavior:  The locations of these vulnerabilities have been corrected, and a new process has been introduced to automatically prevent introduction of new instances.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### False Positive 1
**Reported Vulnerability** - Avoid standard input in TypeScript
**Location** - `codeql/sarif_reporter/src/index.ts:6`
**Reason for Exemption** - Files in the `codeql` directory are used for executing build reports.  They aren't used in the application, and they aren't used to build the application.  Because AppScan does not provide a way to ignore paths which have nothing to do with the application, I can't disable or fix this reported item.

### CODE REVIEW

**Review by saikumar9 — CHANGES_REQUESTED** [review]
* Great job on introducing `CodeQL` automation to find security vulnerabilities. 👏 
* Please review my comments and refactor accordingly.

File: app/controllers/exchanges/hbx_profiles_controller.rb
Comment (TreyE) [reply app/controllers/exchanges/hbx_profiles_controller.rb:917]: I'll pull in the fix from there so there's no code conflict.

File: .github/workflows/codeql_security_scan.yml
Comment (polographer) [inline .github/workflows/codeql_security_scan.yml:18]: the big question here is why we are custom installing code ql and not using the official [Github Action](https://github.com/github/codeql-action), the way that is done here has disadvantages;
1. we are hardcoding a version
2. we have to maintain it
3. we will lose the ability to see the security issues directly on Github
4. This will create a burden on the developer to create the report and share it with DM if needed

We already have a prototype for Enroll that uses this action on https://github.com/ideacrew/enroll/compare/trunk...test_codeql, and you can browse to the security tab using this branch. 

Since the GHA implementation is cleaner and less prone to errors, I don't think we should merge this as it is.

**Review by polographer — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql_security_scan.yml
Comment (polographer) [reply .github/workflows/codeql_security_scan.yml:18]: 1. it doesn't go to "issues" it goes to the security tab, and it's very easy to track who disabled and why (because it asks mandatory for a reason, I just tested it, and you can search with: `is:closed branch:test_codeql `
2. on the web it shows the item, with the snippet of code, and adds the recommendation and example. 
3. A developer not added to the organization cannot see the security items this is a small gap because we have to add DM to the repo. But it's way easier than having to generate the report locally and then share it with the DM

Comment (TreyE) [reply .github/workflows/codeql_security_scan.yml:18]: After working with the CodeQL stuff you've set up I hard agree - I'm going to pull out the custom install and running my own DB/Analysis build and just use the github action helpers to do it.  I'll let you know once I merge in the fix.

**Review by saikumar9 — DISMISSED** [review]
LGTM!

File: codeql.yml
Comment (polographer) [inline codeql.yml:5]: do we need to add project_gems? we have some baggage there

**Review by polographer — DISMISSED** [review]

File: codeql.yml
Comment (TreyE) [reply codeql.yml:5]: Good catch, we should look into that.

Comment (polographer) [reply codeql.yml:5]: btw: I forgot to add but I don't see it as a blocker, we can merge it as it is and add it later

**Review by saikumar9 — APPROVED** [review]

**Review by polographer — APPROVED** [review]

**Pull request merged**
